api/types/networking.go

```diff
@@ -26,6 +26,13 @@ import (
 	"github.com/gravitational/teleport/api/defaults"
 )
 
+type webContextKey string
+
+const (
+	ClientSrcAddrContextKey webContextKey = "teleport-clientSrcAddrContextKey"
+	ClientDstAddrContextKey webContextKey = "teleport-clientDstAddrContextKey"
+)
```

Comment:
We always set and fetch both at the same time; I'd replace this with a struct that holds both `ClientSrcAddr` and `ClientDstAddr`.

The general consensus in the Go community seems to be that it's preferrable to write dedicated functions to append values to and extract values from a context, and I think that having `ClientAddrContext(context.Context, net.Addr, net.Addr) context.Context` and `ClientAddrFromContext(context.Context) (src net.Addr, dst net.Addr)` functions would in fact make this more convenient to use, instead of having to remember the correct typecast with `context.Value`.

Reply:
I've added helper functions 👍 (don't think struct is needed in that case)